### PR TITLE
Fix ONNX PTQ API memory explosion

### PR DESCRIPTION
**Pull request by vinnamkim**
### Changes
 - Add try-catch statement to `ONNNXModelNormalizer`.
 - Fix statistics collection logic not to accumulate all intermediate tensors.
 - Change from `ResNet101-DUC-7` to `ResNet101-DUC-12`

### Reason for changes
 - Without try-catch statement, it terminates the entire PTQ procedure.
 - Accumulating all intermediate tensors makes memory exploded.

### Related tickets
86465

### Tests
precommit/job/ubuntu20_precommit_onnx/21

### CODE REVIEW

File: nncf/experimental/post_training/api/engine.py [hunk 1]
```diff
@@ -12,11 +12,12 @@
...
 from abc import ABC, abstractmethod
+from typing import Callable, Dict, TypeVar
 
-from typing import Dict
-from typing import TypeVar
-from typing import Callable
+from nncf.common.tensor_statistics.collectors import \
+    TensorStatisticCollectorBase
 from nncf.experimental.post_training.api.sampler import Sampler
+from tqdm import tqdm
```
Comment (daniil-lyakhov) [inline nncf/experimental/post_training/api/engine.py:20]: Please move this import at the beginning according to styleguide

File: nncf/experimental/onnx/model_normalizer.py [hunk 2]
```diff
@@ -86,16 +87,23 @@ def add_const_value_infos_to_graph(graph: onnx.GraphProto):
...
+            modified_model = onnx.shape_inference.infer_shapes(modified_model)
+            add_input_from_initializer(modified_model)
+
+            onnx.checker.check_model(modified_model)
+
+            nncf_logger.info(
+                'Successfully converted the model to the opset = {}'.format(modified_model.opset_import[0].version))
+        except (RuntimeError, ConvertError) as e:
+            modified_model = model
+            nncf_logger.warning(f"Failed to convert version. Use the original model. Reason: {e}")
```
Comment (daniil-lyakhov) [inline nncf/experimental/onnx/model_normalizer.py:106]: Is this ok to keep unmodified model? Will it lead to bugs during compression?

**Comment by daniil-lyakhov** [comment]
Please provide cases when this try-catch block is needed

File: nncf/experimental/post_training/api/engine.py (hunk 1 above)
Comment (vinnamkim) [reply nncf/experimental/post_training/api/engine.py]: Resolved.

**Comment by vinnamkim** [comment]
> Please provide cases when this try-catch block is needed

Please see my comment above.

**Comment by vinnamkim** [comment]
@daniil-lyakhov 

I updated two things to keep things clean, as it will be included in the next release.

1. Updated PTQ API results in `README.md` with the latest
2. Removed deprecated `Dockerfile` and updated docker preparation steps in `README.md`.

File: nncf/experimental/onnx/model_normalizer.py (hunk 2 above)
Comment (daniil-lyakhov) [reply nncf/experimental/onnx/model_normalizer.py:106]: What happen if somebody asked for per-channel quantization for a model, which isn't supported opset13?

Comment (vinnamkim) [reply nncf/experimental/onnx/model_normalizer.py:106]: There will be two cases:
1. Model can be converted properly to opset13 using version converter (what our model normalizer do) `https://github.com/onnx/onnx/blob/main/docs/VersionConverter.md`.
2. Model cannot be converted to opset>=13 .

#1 can be handled, but case #2 only allows `per-tensor` quantization.

File: nncf/experimental/post_training/api/engine.py [hunk 3]
```diff
@@ -12,10 +12,12 @@
 """
 
 from abc import ABC, abstractmethod
+from typing import Callable, Dict, TypeVar
 
-from typing import Dict
-from typing import TypeVar
-from typing import Callable
+from tqdm import tqdm
```
Comment (daniil-lyakhov) [inline nncf/experimental/post_training/api/engine.py:17]: Please remove redundant empty line

**Review by daniil-lyakhov — APPROVED** [review]

**Comment by daniil-lyakhov** [comment]
Please rebuild precommit test on latest changes

**Comment by vinnamkim** [comment]
> Please rebuild precommit test on latest changes

job/precommit/job/ubuntu20_precommit_onnx/23 was passed for the latest.

**Pull request merged**